example.py

```diff
@@ -85,41 +84,25 @@ def run_transaction(conn, op, max_retries=3):
                 # This is a retry error, so we roll back the current
                 # transaction and sleep for a bit before retrying. The
                 # sleep time increases for each failed transaction.
-                logging.debug("got error: %s", e)
+                logging.debug(f"got error: {e}")
                 conn.rollback()
                 logging.debug("EXECUTE SERIALIZATION_FAILURE BRANCH")
                 sleep_ms = (2 ** retry) * 0.1 * (random.random() + 0.5)
-                logging.debug("Sleeping %s seconds", sleep_ms)
+                logging.debug(f"Sleeping {sleep_ms} seconds", )
                 time.sleep(sleep_ms)
 
             except psycopg2.Error as e:
-                logging.debug("got error: %s", e)
+                logging.debug(f"got error: {e}")
                 logging.debug("EXECUTE NON-SERIALIZATION_FAILURE BRANCH")
                 raise e
 
         raise ValueError(f"Transaction did not succeed after {max_retries} retries")
 
 
-def test_retry_loop(conn):
-    """
-    Cause a seralization error in the connection.
-
-    This function can be used to test retry logic.
-    """
-    with conn.cursor() as cur:
-        # The first statement in a transaction can be retried transparently on
-        # the server, so we need to add a dummy statement so that our
-        # force_retry() statement isn't the first one.
-        cur.execute("SELECT now()")
-        cur.execute("SELECT crdb_internal.force_retry('1s'::INTERVAL)")
-    logging.debug("test_retry_loop(): status message: %s", cur.statusmessage)
-
-
 def main():
-    opt = parse_cmdline()
-    logging.basicConfig(level=logging.DEBUG if opt.verbose else logging.INFO)
+    logging.basicConfig(level=logging.DEBUG)
 
-    conn = psycopg2.connect(opt.dsn)
+    conn = psycopg2.connect(os.environ['DATABASE_URL'])
```

Comment:
If the env var `DATABASE_URL` is not set, this result in a traceback, instead of giving the user an informative message.

Reply:
Good point. Added some exception handling here for the case where `DATABASE_URL` is not set.